### PR TITLE
Test: add vitest tests for GET and PUT /api/profiles/me routes

### LINKED ISSUE #3: backend: add `vitest` tests for profile GET /me and PUT /me routes
## Summary
Add tests for the existing authenticated profile endpoints in `apps/backend/src/routes/profiles.ts`.

## Tasks
- [ ] create `apps/backend/src/__tests__/profiles.test.ts`.
- [ ] mock `app.prisma` to return a fake user for GET `/api/profiles/me` and assert `displayName` etc.
- [ ] mock `app.prisma.user.update` for PUT `/api/profiles/me` with valid input and assert body returned.
- [ ] confirm failure path for invalid `accentColor` returns 400 and has validation error details.

## Acceptance Criteria
- [ ] asset test coverage for happy and invalid path.
- [ ] `pnpm --filter @devcard/backend test` passes.

### PR DESCRIPTION
**Pull request by Dipti45sktech**
## Summary
Picked up issue #3 to add proper Vitest tests for the authenticated GET and PUT /api/profiles/me endpoints. Reviewed profiles.ts to understand how the user ID is extracted from JWT and how Prisma queries are handled.
Also checked validators.ts, focusing on the accentColor zod validation for correct failure cases.Set up a reusable buildApp() helper that creates a Fastify instance with Prisma and auth mocked using vi.fn(). This avoids needing a real database while keeping tests close to real behavior.
Wrote 5 test cases covering all required scenarios. These include successful profile fetch and update flows. Handled edge cases like 404 for missing user and 409 for duplicate username. Also added a 400 test for invalid accentColor input.
Ensured sensitive fields like provider and providerId are not returned in responses.
All tests pass locally.

Closes #3

## Type of Change
- [ ] Bug fix
- [ ] New feature
- [ ] Refactor (no functional change)
- [ ] UI / Design change
- [x] Tests only
- [ ] Documentation
- [ ] Infrastructure / DevOps
- [ ] Security

## What Changed
- Created `apps/backend/src/__tests__/profiles.test.ts` with full test coverage
- Mocked `app.prisma.user.findUnique`, `findFirst`, and `update` using `vi.fn()`
- Added a `buildApp()` helper that spins up a test Fastify instance with mocked auth and prisma
- Covered GET happy path, GET 404, PUT success, PUT 400 (invalid accentColor), and PUT 409 (duplicate username)

## How to Test
1. Clone the repo and install dependencies with `pnpm install`
2. Run `pnpm --filter @devcard/backend test`
3. All 5 tests should pass with no errors

## Checklist
- [ ] My code follows the project's coding style (`pnpm -r run lint` passes).
- [ ] TypeScript compiles without errors (`pnpm -r run typecheck`).
- [x] I have added or updated tests for the changes I made.
- [x] All tests pass locally (`pnpm -r run test`).
- [ ] I have updated documentation where necessary.
- [x] No new `console.log` or debug statements left in the code.
- [ ] Breaking changes are documented in this PR description.

## Additional Context
Prisma and authentication are fully mocked so no database setup is needed to run these tests. The `buildApp()` helper can be reused in future test files for other routes.

### CODE REVIEW

File: apps/backend/src/__tests__/profiles.test.ts [hunk 1]
```diff
@@ -1,39 +0,0 @@
...
-    // Expected behavior (from profiles.ts lines 54-63):
-    // const existing = await app.prisma.user.findFirst({
-    //   where: {
-    //     username: parsed.data.username,
-    //     NOT: { id: userId },
-    //   },
-    // });
-    // if (existing) {
-    //   return reply.status(409).send({ error: 'Username already taken' });
-    // }
```
Comment (ShantKhatri) [inline apps/backend/src/__tests__/profiles.test.ts:29]: The commit is removing the existing changes, and not change added, no test is added.

Comment (Dipti45sktech) [reply apps/backend/src/__tests__/profiles.test.ts:29]: Really sorry about that @ShantKhatri ! When I was setting up the test file, I deleted the old placeholder file and created the new one, but didn't verify it was actually being tracked by git before committing. So the commit ended up only showing the deletion and not the new file. I caught the mistake, recreated the file with all the tests, verified all 5 pass locally, and pushed the updated commit to the same branch. The test file should now be properly visible in the PR. Really sorry for the confusion, please have a look now!

**Review by ShantKhatri — APPROVED** [review]
